operators/src/plot/temporal_vector_line_plot.rs

```diff
@@ -0,0 +1,360 @@
+                                    .map(|v| v.to_string())
+                                    .ok_or(Error::FeatureDataValueMustNotBeNull)?,
+                                FeatureDataValue::Text(v) => v.clone(),
+                                FeatureDataValue::NullableText(v) => {
+                                    v.ok_or(Error::FeatureDataValueMustNotBeNull)?
+                                }
+                                FeatureDataValue::Category(v) => v.to_string(),
+                                FeatureDataValue::NullableCategory(v) => {
+                                    v.map(|v| v.to_string())
+                                        .ok_or(Error::FeatureDataValueMustNotBeNull)?
+                                }
+                            };
+
+                            let value: f64 = match values.get_unchecked(i) {
+                                FeatureDataValue::Int(v) => v as f64,
+                                FeatureDataValue::NullableInt(v) => v.map_or(0.0, |v| v as f64), // TODO: NAN better default?
+                                FeatureDataValue::Float(v) => v,
+                                FeatureDataValue::NullableFloat(v) => v.unwrap_or(0.0),
+                                FeatureDataValue::Category(v) => f64::from(v),
+                                FeatureDataValue::NullableCategory(v) => v.map_or(0.0, f64::from),
+                                _ => return Err(Error::InvalidFeatureDataType),
+                            };
+
+                            let time = features.time_intervals()[i];
+
+                            acc.add(id, (time, value));
+                        }
+                        Ok(acc)
+                    }
+                    (Err(err), _) | (_, Err(err)) => Err(err),
+                }
+            })
+            .await?;
+
+        let data_points = values.get_data_points();
+        let measurement = Measurement::Unitless; // TODO: attach actual unit if we know it
+        MultiLineChart::new(data_points, measurement)
+            .to_vega_embeddable(false)
+            .context(error::DataType)
+    }
+}
+
+struct TemporalValue {
+    pub time: TimeInterval,
+    pub value: f64,
+}
+
+impl From<(TimeInterval, f64)> for TemporalValue {
+    fn from(value: (TimeInterval, f64)) -> Self {
+        Self {
+            time: value.0,
+            value: value.1,
+        }
+    }
+}
+
+#[derive(Default)]
+struct FeatureAttributeValues<const LENGTH: usize> {
+    values: HashMap<String, Vec<TemporalValue>>,
+}
+
+impl<const LENGTH: usize> FeatureAttributeValues<LENGTH> {
+    pub fn add<V>(&mut self, id: String, value: V)
+    where
+        V: Into<TemporalValue>,
+    {
+        let len = self.values.len();
+
+        match self.values.entry(id) {
+            Occupied(mut entry) => entry.get_mut().push(value.into()),
+            Vacant(entry) => {
+                if len < LENGTH {
+                    entry.insert(vec![value.into()]);
+                }
```

Comment:
So you just ignore it in the other case?
You could either return a Result or doc-comment the add function.

Reply:
dfece38832b7ccb1ba2e73f1de6b619818b93afc